runtests.py

```diff
@@ -491,6 +498,97 @@ def build_project(args):
 
     return site_dir, site_dir_noarch
 
+def asv_compare_config(bench_path, args, h_commits):
+    """
+    Fill the required build options through custom variable
+    'numpy_build_options'
+    """
+    conf_path = os.path.join(bench_path, "asv_compare.conf.json")
+    nconf_path = os.path.join(bench_path, "_asv_compare.conf.json")
+
+    # add custom build
+    build = []
+    if args.parallel > 1:
+        build += ["-j", str(args.parallel)]
+    if args.cpu_baseline:
+        build += ["--cpu-baseline", args.cpu_baseline]
+    if args.cpu_dispatch:
+        build += ["--cpu-dispatch", args.cpu_dispatch]
+    if args.disable_optimization:
+        build += ["--disable-optimization"]
+
+    is_cached = asv_substitute_config(conf_path, nconf_path,
+        numpy_build_options = ' '.join([f'\\"{v}\\"' for v in build]),
+    )
+    if not is_cached:
+        asv_clear_cache(bench_path, h_commits)
+    return nconf_path
+
+def asv_clear_cache(bench_path, h_commits, env_dir="env"):
+    """
+    Force ASV to clear the cache according to specified 'h_commits'
+    """
+    asv_build_pattern = os.path.join(bench_path, env_dir, "*", "asv-build-cache")
```

Comment:
```suggestion
    asv_build_pattern = os.path.join(bench_path, env_dir, env_hash, "asv-build-cache")
```

@seberg, the only downside is here, `asv_clear_cache` clear involved commits cache dir from all environment paths

Reply:
not a big issue I guess.